app/api/threads/list_threads.feature

```diff
@@ -0,0 +1,168 @@
+      | @LaboratoryMember_WithoutApprovedAccessPersonalInfo | FirstName_NotApproved | LastName_NotApproved |
+    And @LaboratoryMember_WithApprovedAccessPersonalInfo is a member of the group @Laboratory who has approved access to his personal info
+    And @LaboratoryMember_WithoutApprovedAccessPersonalInfo is a member of the group @Laboratory
+    And the database has the following table 'items':
+      | id | type | default_language_tag |
+      | 1  | Task | fr                   |
+      | 2  | Task | en                   |
+    And the database has the following table 'items_strings':
+      | item_id | language_tag | title      |
+      | 1       | en           | Beginning  |
+      | 1       | fr           | Debut      |
+      | 2       | en           | Experiment |
+    And the database has the following table 'threads':
+      | item_id | participant_id                                      | status                  | message_count | latest_update_at    | helper_group_id |
+      | 1       | @LaboratoryMember_WithApprovedAccessPersonalInfo    | waiting_for_trainer     | 0             | 2023-01-01 00:00:01 | @Laboratory     |
+      | 2       | @LaboratoryMember_WithoutApprovedAccessPersonalInfo | waiting_for_participant | 1             | 2023-01-01 00:00:02 | @Laboratory     |
+    When I send a GET request to "/threads?watched_group_id=@Laboratory"
+    Then the response code should be 200
+    And the response body should be, in JSON:
+    """
+      [
+        {
+          "item": {
+            "id": "1",
+            "language_tag": "fr",
+            "title": "Debut",
+            "type": "Task"
+          },
+          "latest_update_at": "2023-01-01T00:00:01Z",
+          "message_count": 0,
+          "participant": {
+            "id": "@LaboratoryMember_WithApprovedAccessPersonalInfo",
+            "login": "LaboratoryMember_WithApprovedAccessPersonalInfo",
+            "first_name": "FirstName_Approved",
+            "last_name": "LastName_Approved"
+          },
+          "status": "waiting_for_trainer"
+        },
+        {
+          "item": {
+            "id": "2",
+            "language_tag": "en",
+            "title": "Experiment",
+            "type": "Task"
+          },
+          "latest_update_at": "2023-01-01T00:00:02Z",
+          "message_count": 1,
+          "participant": {
+            "id": "@LaboratoryMember_WithoutApprovedAccessPersonalInfo",
+            "login": "LaboratoryMember_WithoutApprovedAccessPersonalInfo",
+            "first_name": "",
+            "last_name": ""
+          },
+          "status": "waiting_for_participant"
+        }
+      ]
+    """
+
+  Scenario: Should get the threads whose the participant is a descendant of the watched_group_id
+    Given I am @A_UniversityManagerCanWatch
+    When I send a GET request to "/threads?watched_group_id=@A_University"
+    And it should be a JSON array with 2 entries
```

Comment:
The check seems redundant with the fact that the next table has two rows

Reply:
fixed